terraso_backend/apps/graphql/schema/landscape_groups.py

```diff
@@ -41,9 +42,19 @@ def mutate_and_get_payload(cls, root, info, **kwargs):
         called both when adding and updating Landscape Groups. The `kwargs`
         receives a dictionary with all inputs informed.
         """
+        user = info.context.user
+        landscape = Landscape.objects.get(slug=kwargs.pop("landscape_slug"))
+        group = Group.objects.get(slug=kwargs.pop("group_slug"))
+
+        ff_check_permission_on = settings.FEATURE_FLAGS["CHECK_PERMISSIONS"]
+        user_has_delete_permission = user.has_perm(LandscapeGroup.get_perm("add"), obj=landscape.pk)
```

Comment:
suggestion(non-blocking): Maybe the `user_has_delete_permission` only needs to exist if the feature flag is true to avoid that computation if the feature is not enabled